File: server/controller/ProjectDb.go
package controller

import (
	"code-generator-go/server/common"
	"code-generator-go/server/model"
	"net/http"
)

func ProjectDbSelect(w http.ResponseWriter, r *http.Request) {
	projectId := r.FormValue("projectId")
	id := r.FormValue("id")
	data := projectDbDao.Select(projectId, id)
	common.SuccessData(w, data)
}

func ProjectDbInsert(w http.ResponseWriter, r *http.Request) {
	temp := model.ProjectDb{}
	common.Bind(r, &temp)
	row := projectDbDao.Insert(temp.ProjectId, temp)
	if !row {
		common.FailMsg(w, "Save ProjectDb Failed")
		return
	}
	common.SuccessMsg(w, "Save ProjectDb Success")
}

func ProjectDbUpdate(w http.ResponseWriter, r *http.Request) {
	temp := model.ProjectDb{}
	common.Bind(r, &temp)
	row := projectDbDao.Update(temp.ProjectId, temp.Id, temp)
	if !row {
		common.FailMsg(w, "Update ProjectDb Failed")
		return
	}
	common.SuccessMsg(w, "Update ProjectDb Success")
}

File: web/src/pages/code-generator/api/project-db.js
import request from '@/utils/request'

export function projectDbSelect(id) {
  return request({
    url: '/projectDb/select',
    method: 'get',
    params: { projectId: id }
  })
}

export function projectDbInsert(data) {
  return request({
    url: '/projectDb/insert',
    method: 'post',
    data
  })
}

export function projectDbUpdate(data) {
  return request({
    url: '/projectDb/update',
    method: 'put',
    data
  })
}
